### PR TITLE
Fixed error: in eventChannel: subscribe should return a function to unsubscribe 

### PR DESCRIPTION
**Pull request by xxsnakerxx**
Fixed error: in eventChannel: subscribe should return a function to unsubscribe -> [docs](https://yelouafi.github.io/redux-saga/docs/api/index.html#eventchannelsubscribe-buffer-matcher)

`redux-saga@0.11.1` throws the error

### CODE REVIEW

**Comment by jbrodriguez** [comment]
I'm getting this error as well and this PR solved it.
Any chance this will be merged any time soon ?

**Comment by geof90** [comment]
It has merge conflicts, so I am unable to merge it. @xxsnakerxx think you could fix them? Thanks!

**Comment by xxsnakerxx** [comment]
@geof90 Done!

File: index.js [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+    return () => AppState.removeEventListener("change", onAppStateChange);
   });
 }
 
 /**
  * Delays calling sync until the app is ready for it. This allows
- * apps to "throttle" calling sync until after an initial grace/onboarding
- * period, so that end-users are interrupted too soon.
+ * apps to "throttle" calling sync until after an initial onboarding
+ * experience, so that end-users are interrupted too soon.
```
Comment (geof90) [inline index.js]: Could you update this comment?
`...after an initial onboarding time period...`

File: index.js [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
  *
  * @param delayByInterval Number of seconds to delay calling sync
  * @param delayByAction Name of a Redux action to wait for being dispatched before calling sync.
  */
 function* delaySync(delayByInterval, delayByAction) {
-  const key = yield call(AsyncStorage.getItem, codePushDelayKey);
+  const codePushSagaKey = "CODE_PUSH_SAGA_KEY";
```
Comment (geof90) [inline index.js]: I prefer the name codePushDelayKey, but I agree with your change that its scope should be limited to this function

File: index.js [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
-    yield call(AsyncStorage.setItem, codePushDelayKey, codePushDelayValue);
+  if (!key) {
+    yield call(AsyncStorage.setItem, codePushSagaKey, "VALUE");
 
     let delayEvents = {
       interval: call(delay, delayByInterval * 1000)
     };
 
-    // If the consumer specified an action to cancel the delay
-    // period, then add the take effect to the race conditions.
```
Comment (geof90) [inline index.js:43]: Could we keep this comment?

File: index.js [hunk 4]
```diff
@@ -91,9 +91,6 @@ export default function* codePushSaga(options = {}) {
     request: take(options.syncActionName)
   };
 
-  // If the caller requested sync to be triggered
-  // on app resume, then create the event channel
-  // and add it to our race conditions.
```
Comment (geof90) [inline index.js:96]: Could we keep all the comments you deleted?

**Comment by geof90** [comment]
Thanks @xxsnakerxx!

**Pull request merged**